### PR TITLE
Refactor JS API and added check on API with test

### PR DESCRIPTION
**Pull request by infinyte7**
## Purpose / Description
I have created separated class for JS API constants and added check on API which modify contents in AnkiDroid.

Now it will be easier to add new api.

## Fixes
Fixes _Link to the issues._

## Approach
1. Created a separate class containing for api error code and api name
2. Extracted check on api to a function 
3. Added test on `bury` and `suspend` card and notes

## How Has This Been Tested?
Tested using chrome remote debugging and android devices

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
https://developer.android.com/reference/android/webkit/JavascriptInterface

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidJsAPITest.kt [hunk 1]
```diff
@@ -231,6 +231,85 @@ class AnkiDroidJsAPITest : RobolectricTest() {
...
+        // count number of notes
+        assertThat(reviewer.mSched.cardCount(), equalTo(1))
+    }
+
+    private fun createTestScript(apiName: String): String {
+        // create js script for evaluating in webview
+        var script = "javascript:(function () {\n"
+
+        // add js api developer contract
+        script += "var jsApi = {\"version\" : \"0.0.1\", \"developer\" : \"dev@mail.com\"};\n"
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidJsAPITest.kt:302]: A minor point, but when you need a "fake" email address, always use the RFC standard ones:

https://www.techrepublic.com/blog/it-security/use-rfc-2606-example-domains-for-example-emails/
https://datatracker.ietf.org/doc/html/rfc2606#section-3

**Review by mikehardy — CHANGES_REQUESTED** [review]
I think this is a reasonable refactor and incrementally gets us closer to verifying the API calls without all the code in AbstractFlashcardViewer - thanks

Just one minor point that the example email should be safer

File: AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidJsAPITest.kt (hunk 1 above)
Comment (infinyte7) [reply AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidJsAPITest.kt]: Thanks, I learned new things.

**Review by mikehardy — APPROVED** [review]
These LGTM thanks

**Review by david-allison-1 — COMMENTED** [review]
Forgot to hit "comment", sorry

**Comment by infinyte7** [comment]
I will update it later.

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * You should have received a copy of the GNU General Public License along with         *
+ * this program.  If not, see http://www.gnu.org/licenses/>.                            *
+ *                                                                                      *
+ * *************************************************************************************/
+
+package com.ichi2.anki
+
+class AnkiDroidJsAPIConstants {
+    // JS API ERROR CODE
+    @kotlin.jvm.JvmField
```
[the remark this answers is not in the document]
Comment (infinyte7) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt]: I have updated suggested change to `object AnkiDroidJsAPIConstants` and `const val`

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    val ankiJsErrorCodeSuspendCard: Int = 4
+
+    @kotlin.jvm.JvmField
+    val ankiJsErrorCodeBuryNote: Int = 5
+    @kotlin.jvm.JvmField
+    val ankiJsErrorCodeSuspendNote: Int = 6
+
+    // js api developer contact
+    @kotlin.jvm.JvmField
+    var mCardSuppliedDeveloperContact = ""
```
[the remark this answers is not in the document]
Comment (infinyte7) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPIConstants.kt]: The variable belongs to `AnkiDroidJsAPI` so I moved it to that class.

**Review by david-allison-1 — APPROVED** [review]
Awesome!

**Review by mikehardy — APPROVED** [review]
Nice!

**Comment by mikehardy** [comment]
Hi there! November OpenCollective Notice

Apologies for the delay, and if you're already submitted something to OpenCollective for PRs merged in November 2021, you may feel free to ignore this and it will be processed shortly

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for November 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of August)

Thanks!

**Pull request merged**